spec/spec_helper.cr

```diff
@@ -153,7 +154,7 @@ class Crystal::SpecRunOutput
   end
 end
 
-def run(code, filename = nil, inject_primitives = true, debug = Crystal::Debug::None)
+def run(code, filename = nil, inject_primitives = true, debug = Crystal::Debug::None, *, file = __FILE__)
```

Comment:
The specs doesn't care about the location of this files.
They are not even generated sometimes (JIT is used unless prelude is required).
I would avoid adding this named arg and just using a fixed tempfile prefix/sufix in this helper.

Reply:
I'm not sure I understand why you'd want to remove this. Maybe it's not strictly necessary, but using `with_tempfile` here certainly doesn't hurt and it's easier than managing a tempfile location manually.

Reply:
As I said, if specs are failing I would probably run just a few or one. Or even add a debug message for the spec/tempfile association while running specs in some verbose mode.

Having `filename` and `file` arguments in the same method seems misleading.

So, between not been able see an addition here, plus the misleading name, plus having more data to cary on runtime on the high populated spec suite of the compiler, I would vote for removing it.

Reply:
Okay, I'll revert it.